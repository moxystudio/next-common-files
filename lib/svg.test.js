'use strict';

const { withSVG } = require('../index.js');

const options = {
    dev: true,
    isServer: true,
};

const createConfig = () => ({
    module: {
        rules: [],
    },
});

it('should configure a unique rule correctly with inline true', () => {
    const webpackConfig = withSVG({ inline: true })().webpack(createConfig(), options);

    expect(webpackConfig.module.rules).toHaveLength(1);
    expect(webpackConfig.module.rules[0]).toMatchSnapshot();
});

it('should configure a unique rule correctly with inline false', () => {
    const webpackConfig = withSVG({ inline: false })().webpack(createConfig(), options);

    expect(webpackConfig.module.rules).toHaveLength(1);
    expect(webpackConfig.module.rules[0]).toMatchSnapshot();
});

it('should have inline false by default', () => {
    const webpackConfigWithInline = withSVG({ inline: true })().webpack(createConfig(), options);
    const webpackConfigWithFalse = withSVG({ inline: false })().webpack(createConfig(), options);
    const webpackConfigWithDefault = withSVG()().webpack(createConfig(), options);

    expect(webpackConfigWithInline.module.rules).not.toStrictEqual(webpackConfigWithFalse.module.rules);
    expect(webpackConfigWithFalse.module.rules).toStrictEqual(webpackConfigWithDefault.module.rules);
});

it('should return a function if next config is a function', () => {
    const nextConfig = {
        webpack: jest.fn(() => 'foo'),
    };

    const webpackConfig = withSVG()(nextConfig).webpack(createConfig(), options);

    expect(nextConfig.webpack).toHaveBeenCalledTimes(1);
    expect(webpackConfig).toBe('foo');
});
