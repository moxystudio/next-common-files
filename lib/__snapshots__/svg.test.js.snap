// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should configure a unique rule correctly with inline false 1`] = `
Object {
  "test": /\\\\\\.svg\\$/,
  "use": Array [
    Object {
      "loader": "url-loader",
      "options": Object {
        "emitFile": false,
        "limit": 0,
        "name": "images/[name].[ext]",
        "outputPath": "static/chunks/media",
        "publicPath": "_next/static/chunks/media",
      },
    },
    Object {
      "loader": "svgo-loader",
      "options": Object {
        "plugins": Array [
          Object {
            "removeTitle": true,
          },
          Object {
            "removeDimensions": false,
          },
          Object {
            "removeViewBox": false,
          },
          Object {
            "cleanupIDs": false,
          },
        ],
      },
    },
  ],
}
`;

exports[`should configure a unique rule correctly with inline true 1`] = `
Object {
  "test": /\\\\\\.svg\\$/,
  "use": Array [
    Object {
      "loader": "raw-loader",
    },
    Object {
      "loader": "svgo-loader",
      "options": Object {
        "plugins": Array [
          Object {
            "removeTitle": true,
          },
          Object {
            "removeDimensions": false,
          },
          Object {
            "removeViewBox": false,
          },
          Object {
            "cleanupIDs": false,
          },
        ],
      },
    },
    Object {
      "loader": "svg-css-modules-loader",
      "options": Object {
        "transformId": true,
      },
    },
  ],
}
`;
